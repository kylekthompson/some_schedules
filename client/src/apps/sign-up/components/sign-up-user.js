import PropTypes from 'prop-types';
import React from 'react';
import SignUpUserForm from 'apps/sign-up/components/sign-up-user-form';
import styled from 'styled-components';
import { Card, Container, Text } from 'components/style';
import { Separator } from 'components/form';
import { authenticated } from 'components/authentication';

const StyledContainer = styled(Container)`
  align-items: center;
  padding-top: 100px;
`;

const StyledCard = styled(Card)`
  align-items: center;
`;

export function SignUpUser({ requestSignIn }) {
  return (
    <StyledContainer>
      <StyledCard>
        <Text size={20} weight="bold">Sign Up - User</Text>
        <Separator />
        <SignUpUserForm onSuccess={requestSignIn} />
      </StyledCard>
    </StyledContainer>
  );
}

SignUpUser.propTypes = {
  requestSignIn: PropTypes.func.isRequired,
  user: PropTypes.object,
};

export default authenticated(SignUpUser);
