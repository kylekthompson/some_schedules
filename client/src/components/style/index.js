export { default as Box } from 'src/components/style/box';
export { default as Link } from 'src/components/style/link';
export { default as Text } from 'src/components/style/text';
