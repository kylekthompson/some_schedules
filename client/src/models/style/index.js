export { default as colors } from 'src/models/style/colors';
export { default as fonts } from 'src/models/style/fonts';
export { default as shadows } from 'src/models/style/shadows';
