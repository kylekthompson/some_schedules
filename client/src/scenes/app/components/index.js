export { default as Container } from 'scenes/App/components/Container';
export { default as HeaderContainer } from 'scenes/App/components/HeaderContainer';
export { default as HeaderLinks } from 'scenes/App/components/HeaderLinks';
