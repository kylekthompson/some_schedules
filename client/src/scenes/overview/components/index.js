export { default as Container } from 'scenes/Overview/components/Container';
export { default as HeaderLinks } from 'scenes/Overview/components/HeaderLinks';
export { default as Routes } from 'scenes/Overview/components/Routes';
