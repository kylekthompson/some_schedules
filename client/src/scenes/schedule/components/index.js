export { default as Container } from 'scenes/Schedule/components/Container';
export { default as ContentContainer } from 'scenes/Schedule/components/ContentContainer';
export { default as SidebarContainer } from 'scenes/Schedule/components/SidebarContainer';
