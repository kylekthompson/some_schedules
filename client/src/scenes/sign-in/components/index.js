export { default as Container } from 'scenes/sign-in/components/container';
