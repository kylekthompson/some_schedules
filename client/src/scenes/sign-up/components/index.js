export { default as Container } from 'scenes/sign-up/components/container';
