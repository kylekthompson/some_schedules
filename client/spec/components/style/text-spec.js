import React from 'react';
import { Text } from 'components/style';
import { colors } from 'models/style';
import { mount } from 'spec/support/mount';

describe('<Text />', () => {
  describe('without any props', () => {
    it('is a paragraph tag with font size 12px and color black', () => {
      const { container } = mount(<Text />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed a size', () => {
    it('has the specified font size', () => {
      const { container } = mount(<Text size={20} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed a color', () => {
    it('has the specified color', () => {
      const { container } = mount(<Text color={colors.white()} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed a component', () => {
    it('renders that component', () => {
      const { container } = mount(<Text component="span" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed children', () => {
    it('renders the children', () => {
      const { getByText } = mount(<Text>Words</Text>);
      getByText('Words');
    });
  });
});
