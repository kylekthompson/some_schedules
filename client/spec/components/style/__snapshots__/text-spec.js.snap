// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> when passed a color has the specified color 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  color: rgba(255,255,255,1);
}

<p
  class="c0"
  color="rgba(255, 255, 255, 1)"
/>
`;

exports[`<Text /> when passed a component renders that component 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  color: rgba(12,27,42,1);
}

<span
  class="c0"
/>
`;

exports[`<Text /> when passed a size has the specified font size 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-weight: 400;
  font-size: 20px;
  color: rgba(12,27,42,1);
}

<p
  class="c0"
  size="20"
/>
`;

exports[`<Text /> without any props is a paragraph tag with font size 12px and color black 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  color: rgba(12,27,42,1);
}

<p
  class="c0"
/>
`;
